addons/test_mail_full/tests/test_res_users.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.tests.common import MailCommon, mail_new_test_user
+
+
+class TestResUsers(MailCommon):
+
+    def test_deactivate_portal_users(self):
+        """Test that the email and the phone are blacklisted
+        when a portal user deactivate his own account.
+        """
+        portal_user = mail_new_test_user(
+            self.env,
+            login='portal_user',
+            mobile='+32 494 12 34 56',
+            phone='+32 494 12 34 89',
+            password='password',
+            name='Portal User',
+            email='portal@test.example.com',
+            groups='base.group_portal',
+        )
+
+        portal_user_2 = mail_new_test_user(
+            self.env,
+            login='portal_user_2',
+            mobile='+32 494 12 34 22',
+            phone='invalid phone',
+            password='password',
+            name='Portal User 2',
+            email='portal_2@test.example.com',
+            groups='base.group_portal',
+        )
+
+        # Remove existing blacklisted email / phone (they will be sanitized, so we avoid to sanitize them here)
+        self.env['mail.blacklist'].search([]).unlink()
+        self.env['phone.blacklist'].search([]).unlink()
+
+        (portal_user | portal_user_2)._deactivate_portal_user()
+
+        self.assertFalse(portal_user.active, 'Should have archived the user')
```

Comment:
Unless they give valuable information, I would not include the string at the end.